jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -260,14 +265,22 @@ private Image pullBaseImage(
   }
 
   /**
-   * Looks through a manifest list for any amd64/linux manifest and downloads and returns the first
-   * manifest it finds.
+   * Looks through a manifest list for the user specified arch/os manifest and downloads and returns
+   * the first manifest it finds.
    */
-  private ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
+  @VisibleForTesting
+  ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
       RegistryClient registryClient, V22ManifestListTemplate manifestListTemplate)
       throws RegistryException, IOException {
 
-    List<String> digests = manifestListTemplate.getDigestsForPlatform("amd64", "linux");
+    if (buildContext.getContainerConfiguration() != null) {
+      Set<Platform> platforms = buildContext.getContainerConfiguration().getPlatforms();
+      Platform platform = platforms.stream().findFirst().get();
+      architecture = platform.getArchitecture();
+      os = platform.getOs();
+    }
```

Comment:
I find no reason to declare these as method fields.

Reply:
Thats the only way we can access the variable inside the platform object.

Reply:
What I meant is basically this:
```java
String architecture = "amd64"; // no need to have them as method fields, just local variables
String os = "linux";
if (...) {
  ...
  architecture = platform.getArchitecture();
  os = platforms.getOs();
}
List<String> digests = manifestListTemplate.getDigestsForPlatform(architecture, os);
```

Even with this, it's still not ideal because this `PullBaseImageStep` is yet another entity having an opinion of what a default should be. But this issue will be gone once #2619 is merged.